addons/medical_treatment/XEH_postInit.sqf

```diff
@@ -21,3 +21,60 @@ if (isServer) then {
 [QGVAR(actionCheckPulseLocal), FUNC(actionCheckPulseLocal)] call CBA_fnc_addEventHandler;
 [QGVAR(actionCheckBloodPressureLocal), FUNC(actionCheckBloodPressureLocal)] call CBA_fnc_addEventHandler;
 [QGVAR(actionPlaceInBodyBag), FUNC(actionPlaceInBodyBag)] call CBA_fnc_addEventHandler;
+
+// Notification lets the player known if they're being treated
+[QGVAR(initiated), {
+    params ["_caller", "_target"];
+
+    // Notifications don't make sense for self-healing or when dead
+    if (_target != ACE_player || {_target == _caller} || {!alive _target}) exitWith {};
+
+    // Can't tell who is providing aid when you're unconscious
+    private _name = if IS_UNCONSCIOUS(_target) then {
+        LLSTRING(UnknownPerson);
+    } else {
+        name _caller
+    };
+
+    QGVAR(notification) cutText [format [LLSTRING(Notification), _name], "PLAIN DOWN"];
+
+    // We track who is currently treating the unit
+    private _responders = _target getVariable [QGVAR(responders), []];
+    _responders pushBack _caller;
+    _target setVariable [QGVAR(responders), _responders];
+}] call CBA_fnc_addEventHandler;
+
+// Notification may no longer apply when treatment ends
+private _treatmentEnded = {
+    params ["_caller", "_target"];
+
+    // Notifications don't make sense for self-healing or when dead
+    if (_target != ACE_player || {_target == _caller} || {!alive _target}) exitWith {};
+
+    // Remove from tracking (in place)
+    private _responders = _target getVariable [QGVAR(responders), []];
+    _responders deleteAt (_responders find _caller);
```

Comment:
This logic is missing an edge case for handling `_responders` items that have never raised an event for `_treatmentEnded`; i.g. on network issues. Do we wish to handle this here as well?

Reply:
Yeah this is something we definitely want to handle, I've assumed that the medic being killed mid-treatment would trigger the treatment failure condition, but we probably need to filter out any `null` units too for disconnections.

I'm not entirely sure how best to handle a player disconnection who is then taken over by an AI (as they'd then no longer be treating).

Reply:
We'd probably have to integrate a timeout for it?

Reply:
I kind of hate that it makes it more complicated for such an edge case, but I can't think of a better way to handle that scenario

Reply:
🤔 Actually we could store the value of `EFUNC(common,isPlayer)` and then check if it's still true in future

Reply:
Actually actually, it looks like `medical_ai` is written such that AI never treat player controlled characters, so we could just check if the unit is still player controlled

Reply:
Perfect. If we need to have this work for AI as well, we can always implement a timeout based system around this.